File: jdbc-examples/src/main/java/io/vertx/example/jdbc/simple/JDBCExample.java
package io.vertx.example.jdbc.simple;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.Handler;
import io.vertx.core.json.JsonArray;
import io.vertx.core.json.JsonObject;
import io.vertx.example.util.Runner;
import io.vertx.ext.jdbc.JDBCClient;
import io.vertx.ext.sql.ResultSet;
import io.vertx.ext.sql.SQLConnection;

/*
 * @author <a href="mailto:pmlopes@gmail.com">Paulo Lopes</a>
 */
public class JDBCExample extends AbstractVerticle {

  // Convenience method so you can run it in your IDE
  public static void main(String[] args) {
    Runner.runExample(JDBCExample.class);
  }

  @Override
  public void start() throws Exception {

    final JDBCClient client = JDBCClient.createShared(vertx, new JsonObject()
        .put("url", "jdbc:hsqldb:mem:test?shutdown=true")
        .put("driver_class", "org.hsqldb.jdbcDriver")
        .put("max_pool_size", 30));

    client.getConnection(conn -> {
      if (conn.failed()) {
        System.err.println(conn.cause().getMessage());
        return;
      }

      // create a test table
      execute(conn.result(), "create table test(id int primary key, name varchar(255))", create -> {
        // insert some test data
        execute(conn.result(), "insert into test values(1, 'Hello')", insert -> {
          // query some data
          query(conn.result(), "select * from test", rs -> {
            for (JsonArray line : rs.getResults()) {
              System.out.println(line.encode());
            }

            // and close the connection
            conn.result().close(done -> {
              if (done.failed()) {
                throw new RuntimeException(done.cause());
              }
            });
          });
        });
      });
    });
  }

  private void execute(SQLConnection conn, String sql, Handler<Void> done) {
    conn.execute(sql, res -> {
      if (res.failed()) {
        throw new RuntimeException(res.cause());
      }

      done.handle(null);
    });
  }

  private void query(SQLConnection conn, String sql, Handler<ResultSet> done) {
    conn.query(sql, res -> {
      if (res.failed()) {
        throw new RuntimeException(res.cause());
      }

      done.handle(res.result());
    });
  }
}

File: jdbc-examples/src/main/kotlin/io/vertx/example/jdbc/simple/JDBCExample.kt
package io.vertx.example.jdbc.simple

import io.vertx.ext.jdbc.JDBCClient
import io.vertx.kotlin.core.json.*

class JDBCExample : io.vertx.core.AbstractVerticle()  {
  override fun start() {

    var client = JDBCClient.createShared(vertx, json {
      obj(
        "url" to "jdbc:hsqldb:mem:test?shutdown=true",
        "driver_class" to "org.hsqldb.jdbcDriver",
        "max_pool_size" to 30,
        "user" to "SA",
        "password" to ""
      )
    })

    client.getConnection({ conn ->
      if (conn.failed()) {
        System.err.println(conn.cause().getMessage())
        return
      }

      var connection = conn.result()
      connection.execute("create table test(id int primary key, name varchar(255))", { res ->
        if (res.failed()) {
          throw java.lang.RuntimeException(res.cause())
        }
        // insert some test data
        connection.execute("insert into test values(1, 'Hello')", { insert ->
          // query some data
          connection.query("select * from test", { rs ->
            for (line in rs.result().results) {
              println(line.toString())
            }

            // and close the connection
            connection.close({ done ->
              if (done.failed()) {
                throw java.lang.RuntimeException(done.cause())
              }
            })
          })
        })
      })
    })
  }
}
